src/hu_nmt/data_augmentator/base/depedency_parser_base.py

```diff
@@ -78,6 +86,9 @@ def _sentence_process_batch_to_node_relationship_list(process_batch: SentencePro
             -> List[List[NodeRelationship]]:
         raise NotImplementedError
 
+    def tokenize(self, sentence: str):
+        return self.nlp_pipeline(sentence)
```

Comment:
I found it confusing to add a `tokenize` method to a dependencyParser class. Also because it just runs an nlp_pipeline. Maybe we should rename all these dependencyParser classes to something like nlpPipeline, because these classes now have two main functions (to tokenize and to produce the POS & dep parsing enrichments)

Reply:
so actually the tokenize method is good here, let's just think about renaming these classes. What do you think?

Reply:
Yes, I thought about renaming the class, but I did not have any idea. Should it be NlpPipeline?